src/tools/clippy/tests/ui/crashes/mut_mut_macro.rs

```diff
@@ -14,7 +14,7 @@
 const BAA: *const i32 = 0 as *const i32;
 static mut BAR: *const i32 = BAA;
 static mut FOO: *const i32 = 0 as *const i32;
-static mut BUH: bool = 42.0 < f32::NAN;
+//static mut BUH: bool = 42.0 < f32::NAN;
```

Comment:
Should this be removed?

Reply:
I don't know. This clippy test is weird, it's technically broken (see the comment above), but this line was checking that the `nan_cmp` lint did not fire in a const context, because the `is_nan` method is unstable in const context, which I don't think is a good enough reason to not warn.

Reply:
Since this is not a clippy lint any more, removing this line should be ok, except if clippy team tells you otherwise.
The const context question is relevant. Could you add the const context case in the rustc ui test?

Reply:
Yeah, should be fine. So as requested removed the line and added a const context test case.